src/ClassTypeDetector.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TomasVotruba\UnusedPublic;
+
+use PhpParser\Comment\Doc;
+use PhpParser\Node\Stmt\ClassMethod;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Reflection\Php\PhpMethodReflection;
+use PHPStan\Reflection\ResolvedMethodReflection;
+
+final class ClassTypeDetector
+{
+    public function isTestClass(ClassReflection $classReflection): bool
```

Comment:
after this PR, it will also be easier to integrate other test-frameworks, which use different base-classes.

just a matter of adjusting a single method in the repo now.